go1.18: tls unit tests don't pass locally

### Which jobs are failing?

unit tests

### Which tests are failing?

```
--- FAIL: TestTLSConfig (0.05s)
    webhook_test.go:472: client does not trust server: unexpected error message mismatch:
          Expected: Get .*: x509: certificate signed by unknown authority
          Actual:   Get "https://localhost:64695/?timeout=30s": x509: “webhook_tests_server” certificate is not standards compliant
    webhook_test.go:472: webhook does not support insecure servers: unexpected error message mismatch:
          Expected: Get .*: x509: certificate signed by unknown authority
          Actual:   Get "https://localhost:64705/?timeout=30s": x509: “webhook_tests_server” certificate is not standards compliant
FAIL
FAIL	k8s.io/apiserver/pkg/util/webhook	2.001s
```

### Since when has it been failing?

since switching to go1.18

### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

Weirdly, the test is passing in CI (https://testgrid.k8s.io/sig-release-master-blocking#ci-kubernetes-unit&include-filter-by-regex=TestTLSConfig) and failing locally

### Relevant SIG(s)

/sig api-machinery auth testing

On linux it works fine:
```
$ uname -a && go version && go test k8s.io/apiserver/pkg/util/webhook
Linux bigbox 4.15.0-159-generic #167-Ubuntu SMP Tue Sep 21 08:55:05 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
go version go1.18 linux/amd64
ok  	k8s.io/apiserver/pkg/util/webhook	1.673s
```

on MacOS, it fails:
```
[dims@dims-m1-7029 13:48] ~/go/src/k8s.io/kubernetes ⟩ uname -a && go version && go test k8s.io/apiserver/pkg/util/webhook
Darwin dims-m1-7029.local 21.4.0 Darwin Kernel Version 21.4.0: Mon Feb 21 20:35:58 PST 2022; root:xnu-8020.101.4~2/RELEASE_ARM64_T6000 arm64
go version go1.18 darwin/arm64
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55807: tls: failed to verify client certificate: x509: certificate signed by unknown authority
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55811: remote error: tls: bad certificate
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55813: remote error: tls: bad certificate
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55821: remote error: tls: bad certificate
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55823: remote error: tls: bad certificate
I0328 13:50:14.198620   62336 server_cert_deprecations.go:62] invalid certificate detected while connecting to "localhost": relies on a legacy Common Name field instead of the SAN extension for subject validation
I0328 13:50:14.201034   62336 server_cert_deprecations.go:62] invalid certificate detected while connecting to "localhost": uses an insecure SHA-1 signature
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55825: remote error: tls: bad certificate
I0328 13:50:14.203451   62336 server_cert_deprecations.go:62] invalid certificate detected while connecting to "localhost": uses an insecure SHA-1 signature
2022/03/28 13:50:14 http: TLS handshake error from 127.0.0.1:55827: remote error: tls: bad certificate
--- FAIL: TestTLSConfig (0.04s)
    webhook_test.go:488: client does not trust server: unexpected error message mismatch:
          Expected: Get .*: x509: certificate signed by unknown authority
          Actual:   Get "https://localhost:55810/?timeout=30s": x509: “webhook_tests_server” certificate is not standards compliant
    webhook_test.go:488: webhook does not support insecure servers: unexpected error message mismatch:
          Expected: Get .*: x509: certificate signed by unknown authority
          Actual:   Get "https://localhost:55820/?timeout=30s": x509: “webhook_tests_server” certificate is not standards compliant
FAIL
FAIL	k8s.io/apiserver/pkg/util/webhook	1.835s
FAIL
```

--- a/staging/src/k8s.io/apimachinery/pkg/util/proxy/dial_test.go
+++ b/staging/src/k8s.io/apimachinery/pkg/util/proxy/dial_test.go
@@ -26,7 +26,7 @@ import (
 	"net/http/httptest"
 	"net/url"
 	"reflect"
-	"strings"
+	"regexp"
 	"testing"
 
 	"k8s.io/apimachinery/pkg/util/diff"
@@ -56,7 +56,7 @@ func TestDialURL(t *testing.T) {
 		},
 		"secure, no roots": {
 			TLSConfig:   &tls.Config{InsecureSkipVerify: false},
-			ExpectError: "unknown authority",
+			ExpectError: "unknown authority|not trusted",
 		},
 		"secure with roots": {
 			TLSConfig: &tls.Config{InsecureSkipVerify: false, RootCAs: roots},
@@ -76,7 +76,7 @@ func TestDialURL(t *testing.T) {
 		"secure, no roots, custom dial": {
 			TLSConfig:   &tls.Config{InsecureSkipVerify: false},
 			Dial:        d.DialContext,
-			ExpectError: "unknown authority",
+			ExpectError: "unknown authority|not trusted",
 		},
 		"secure with roots, custom dial": {
 			TLSConfig: &tls.Config{InsecureSkipVerify: false, RootCAs: roots},
@@ -154,7 +154,7 @@ func TestDialURL(t *testing.T) {
 				if tc.ExpectError == "" {
 					t.Errorf("%s: expected no error, got %q", k, err.Error())
 				}
-				if !strings.Contains(err.Error(), tc.ExpectError) {
+				if tc.ExpectError != "" && !regexp.MustCompile(tc.ExpectError).MatchString(err.Error()) {
 					t.Errorf("%s: expected error containing %q, got %q", k, tc.ExpectError, err.Error())
 				}
 				return

--- a/staging/src/k8s.io/apiserver/pkg/util/webhook/webhook_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/util/webhook/webhook_test.go
@@ -50,7 +50,7 @@ const (
 	errBadCertificate    = "Get .*: remote error: tls: bad certificate"
 	errNoConfiguration   = "invalid configuration: no configuration has been provided"
 	errMissingCertPath   = "invalid configuration: unable to read %s %s for %s due to open %s: .*"
-	errSignedByUnknownCA = "Get .*: x509: certificate signed by unknown authority"
+	errSignedByUnknownCA = "Get .*: x509: .*(unknown authority|not standards compliant|not trusted)"
 )
 
 var (

